libindy/src/commands/metrics.rs

```diff
@@ -42,39 +46,112 @@ impl MetricsCommandExecutor {
 
     fn collect(&self) -> IndyResult<String> {
         trace!("_collect >>>");
-        let mut metrics_map= serde_json::Map::new();
-        self.append_threapool_metrics(&mut metrics_map);
-        self.append_wallet_metrics(&mut metrics_map);
-        self.metrics_service.append_command_metrics(&mut metrics_map);
+        let mut metrics_map = serde_json::Map::new();
+        self.append_threapool_metrics(&mut metrics_map)?;
+        self.append_wallet_metrics(&mut metrics_map)?;
+        self.metrics_service
+            .append_command_metrics(&mut metrics_map)?;
         let res = serde_json::to_string(&metrics_map)
-            .to_indy(IndyErrorKind::InvalidState,"Can't serialize a metrics map")?;
+            .to_indy(IndyErrorKind::InvalidState, "Can't serialize a metrics map")?;
 
         trace!("_collect <<< res: {:?}", res);
         debug!("collecting metrics from command thread");
         Ok(res)
     }
 
-    fn append_threapool_metrics(&self, metrics_map: &mut Map<String, Value>) {
+    fn append_threapool_metrics(&self, metrics_map: &mut Map<String, Value>) -> IndyResult<()> {
+        #[derive(Serialize, Deserialize)]
+        struct MetricsTags {
+            label: String,
+        }
+
         let tp_instance = crate::commands::THREADPOOL.lock().unwrap();
-        metrics_map.insert(String::from(THREADPOOL_ACTIVE_COUNT),
-                           Value::from(tp_instance.active_count()));
-        metrics_map.insert(String::from(THREADPOOL_QUEUED_COUNT),
-                           Value::from(tp_instance.queued_count()));
-        metrics_map.insert(String::from(THREADPOOL_MAX_COUNT),
-                           Value::from(tp_instance.max_count()));
-        metrics_map.insert(String::from(THREADPOOL_PANIC_COUNT),
-                           Value::from(tp_instance.panic_count()));
-    }
+        let mut threadpool_threads_count = Vec::new();
+        let mut tag = HashMap::<String, String>::new();
+
+        tag.insert(String::from("label"), String::from(THREADPOOL_ACTIVE_COUNT));
+        threadpool_threads_count.push(
+            serde_json::to_value(MetricsValue::new(tp_instance.active_count(), tag.clone()))
+                .to_indy(IndyErrorKind::IOError, "Unable to convert json")?,
+        );
```

Comment:
Can we simplify this somehow? Like extract common parts of it to common function?